### PR TITLE
chore(orchestrator): implement the WorkflowViewerFormatter

### PR DESCRIPTION
**Pull request by batzionb**
implement the WorkflowViewerFormatter - a utility for converting the WorkflowViewer backend data to data the components can consume
includes a generic Formatter interface for reuse

### CODE REVIEW

File: plugins/orchestrator/src/dataFormatters/DataFormatterInterface.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+interface DataFormatterInterface<Data, FormattedData> {
```
Comment (caponetto) [inline plugins/orchestrator/src/dataFormatters/DataFormatterInterface.ts:1]:
```suggestion
interface DataFormatter<Data, FormattedData> {
```
I think the `Interface` suffix is redundant here. Also applies to the filename.

File: plugins/orchestrator/src/dataFormatters/WorkflowOverviewFormatter.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  name: string;
+  lastTriggered: string;
+  lastRunStatus: string;
+  type: string;
+  avgDuration: string;
+  description: string;
+  format: WorkflowFormat;
+}
+
+export interface FormattedWorkflowOverview {}
```
Comment (caponetto) [inline plugins/orchestrator/src/dataFormatters/WorkflowOverviewFormatter.ts:24]: Duplicated interface?

**Review by caponetto — COMMENTED** [review]
Nice! Thanks!

File: plugins/orchestrator/src/dataFormatters/WorkflowOverviewFormatter.ts (hunk 2 above)
Comment (batzionb) [reply plugins/orchestrator/src/dataFormatters/WorkflowOverviewFormatter.ts]: c2c7bf23ae80c3b82e31b92efc2c2214013c0c78

File: plugins/orchestrator/src/dataFormatters/DataFormatter.ts [hunk 3]
```diff
@@ -0,0 +1,5 @@
+interface DataFormatter<Data, FormattedData> {
+  format(Data): FormattedData;
```
Comment (caponetto) [inline plugins/orchestrator/src/dataFormatters/DataFormatter.ts:2]:
```suggestion
  format(data: Data): FormattedData;
```
Otherwise the CI will fail.

**Review by caponetto — APPROVED** [review]

**Pull request merged**